samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/RefAccessor.java

```diff
@@ -17,22 +17,41 @@
  * the state itself is available as a public property on this class. The reason its here is
  * because the caller of the constructor could pass in null for the state, in which case
  * the factory provided on the GetAsync call will be used.
+ * @param <T> The value type of the RefAccessor class.
  */
-public class RefAccessor<T extends Class> implements StatePropertyAccessor<T> {
+public class RefAccessor<T> implements StatePropertyAccessor<T> {
     private T value;
 
+    /**
+     * Sets the value object.
+     * @param withValue The specified new value.
+     */
     public RefAccessor(T withValue) {
         value = withValue;
     }
 
+    /**
+     * Gets the value object.
+     * @return The value object.
+     */
     public T getValue() {
         return value;
     }
 
+    /**
+     * Gets the TypeName of the value's class.
+     * @return The String representing the TypeName of the value's class.
+     */
     public String getName() {
-        return value.getTypeName();
+        return value.getClass().getTypeName();
```

Comment:
Do you know an example of the value here?

Reply:
I'll debug it and I'll tell you